### PR TITLE
Added Client Type feature flag to begin client type work

### LINKED ISSUE #26346: Add a feature flag to start the Client Type work behind
### Description

In order to begin work on [Client Types](https://github.com/keycloak/keycloak/issues/22262), we need to have an experimental feature flag in place.

### PR DESCRIPTION
**Pull request by vickeybrown**
Closes #26346

### CODE REVIEW

File: js/apps/admin-ui/src/utils/useIsFeatureEnabled.ts
Comment (jonkoops) [inline js/apps/admin-ui/src/utils/useIsFeatureEnabled.ts:1]: There are some formatting issues here that the linter is reporting. You can fix these by running `pnpm --filter admin-ui lint -- --fix` in the `js` directory.

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by jonkoops — DISMISSED** [review]
Not sure about introducing a profile without a feature, but the code looks good to me.

**Comment by jonkoops** [comment]
@vickeybrown you'll have to [sign off](https://github.com/keycloak/keycloak/pull/26389/checks?check_run_id=20739032845) on your commits.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Please update HelpCommandDistTest expected messages in https://github.com/keycloak/keycloak/tree/main/quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help. See eg. #23977 for an example of adding an experimental feature flag

**Comment by hmlnarik** [comment]
@mposolda Since this feature belongs to @keycloak/core  team, could you please review as well?

**Review by jonkoops — DISMISSED** [review]

**Review by mposolda — APPROVED** [review]
@vickeybrown Thanks! I am fine with introducing the experimental feature flag for client types as long as tests are OK.

@hmlnarik PR is assigned to you, so leaving to you to further review/merge whenever you are ok. I've re-triggered the tests (as I am not sure those are real failures or instability)

**Review by jonkoops — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**